python/verticox/likelihood.py

```diff
@@ -188,23 +229,20 @@ def minimize_newton_raphson(x_0, func, jacobian, hessian, eps) -> ArrayLike:
         x_0:
         jacobian:
         hessian:
+        params:
         eps:
 
     Returns:
 
     """
     x = x_0
-    current_jac = jacobian(x)
+    current_jac = jacobian(x, params)
     while np.linalg.norm(current_jac) > eps:
         # logger.debug(f'Old x: {old_x}')
         # logger.debug(f'new x: {x}')
 
-        current_hess = hessian(x)
-        current_jac = jacobian(x)
-
-        x = x - np.matmul(np.linalg.inv(current_hess), current_jac)
+        current_hess = hessian(x, params)
+        current_jac = jacobian(x, params)
 
-        _logger.debug(f'Jacobian: {current_jac}')
-        _logger.debug(f'Norm of jacobian: {np.linalg.norm(current_jac)}')
-        _logger.debug(f'Lz_inner: {func(x)}')
+        x = x - np.linalg.inv(current_hess) @ current_jac
```

Comment:
Usually calculating inverses of matrices is very slow and avoided as it's often not necessary. Especially here because it's in the inner loop. Doing the linear solve is probably a lot more efficient. You could maybe try rewriting the line to use `numpy.linalg.solve()` instead and see if that makes it faster.

Reply:
For some reason the `np.linalg.inv` is wayy faster here than the `np.linalg.solve`. It really puzzles me. We can maybe discuss it when you're working.

Reply:
Very strange. I have found some (older) issues stating this exact problem though, with many people surprised that numpy solve would be much slower than calculating the inverse. I did find that the scipy docs advise you to use `solve` over `inv`, although they claim that `scipy.linalg` is faster than `numpy.linalg` due to always using BLAS. Not sure if that info is still in-date though.

Reply:
Could be worth trying the scipy version of `linalg.solve()` and seeing if it helps